navigator_helpers/pipelines/text_to_code.py

```diff
@@ -115,18 +166,20 @@ def run(
         )
 
         with logging_redirect_tqdm():
-            for _ in range(num_samples):
-                domain, topic, complexity = tags.sample()
-                record = self.tasks.create_record(
-                    domain=domain,
-                    topic=topic,
-                    complexity=complexity,
-                    llm_as_a_judge=self.config.llm_as_a_judge,
-                    syntax_validation=self.config.syntax_validation,
-                    progress_bar=pbar,
-                )
-                synthetic_dataset.append(record)
-                self._save_artifact("synthetic_dataset", synthetic_dataset)
+            with ThreadPoolExecutor(max_workers=max_workers) as executor:
```

Comment:
@pimlock @johnnygreco I am debating to leave this part out or keep it in. What do you think?